### DISCUSSION TITLE
part-ii/2017-100/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#422] [Question comments]
Part II 2017 Number Theory: Paper 1, Section $I$, G 

https://questions.tripos.org/part-ii/2017-100/

**Comment by walrus1729** [comment]
<details>
<summary>Solution</summary>

The first part is bookwork.

For the second part, we have $2^m \equiv -1 \pmod{p}$ and $2^{2m} \equiv +1 \pmod{p}$, so (since $m$ is a power of $2$), $2$ has order $2m$ $\pmod{p}$. Since $2^{p-1} \equiv 1 \pmod{p}$ by Fermat's Little Theorem, we conclude that $2m$ divides $p-1$, so $p \equiv 1 \pmod{2m}$. 

To finish, we use our result from the previous part that if $p \equiv 1 \pmod{8}$, then $2$ is a quadratic residue $\pmod{p}$. Since $m \geq 4$ is a power of $2$, and $p \equiv 1 \pmod{2m}$, we indeed have that $p \equiv 1 \pmod{8}$, so $2 \equiv y^2 \pmod{p}$ for some $y$. Then $2^m \equiv y^{2m} \equiv -1 \pmod{p}$ and $2^{2m} \equiv y^{4m} \equiv +1 \pmod{p}$, so (again since $m$ is a power of $2$), $y$ has order $4m$ $\pmod{p}$. Then, as above, since the order of $y$ $\pmod{p}$ must divide $p-1$ by Fermat's Little Theorem, we conclude that $p \equiv 1 \pmod{4m}$ as required.

</details>
